apps/dashboard/app/controllers/files_controller.rb

```diff
@@ -161,8 +162,35 @@ def edit
 
   private
 
-  def normalized_path
-    Pathname.new("/" + params[:filepath].chomp("/"))
+  def normalized_path(path = params[:filepath])
+    Pathname.new("/" + path.to_s.chomp("/").delete_prefix("/"))
+  end
+
+  def parse_path(path = params[:filepath])
+    match = path.to_s.match(/^(?<remote>[0-9A-Za-z_\.\- ]+:(\/)?)?(?<path>.*)$/)
+    if !::Configuration.files_app_remote_files? || match[:remote].nil?
+      PosixFile.new(normalized_path(path))
+    else
+      remote = match[:remote].chomp("/").chomp(":")
+      path = Pathname.new("/" + match[:path].chomp("/"))
+      RemoteFile.new(path, remote)
+    end
+  end
+
+  def validate_path!
+    if posix_file?
+      AllowlistPolicy.default.validate!(@path)
+    elsif @path.remote_type.nil?
+      raise "Remote does not exist"
```

Comment:
```suggestion
      raise StandardError, "Remote '#{remote}' does not exist"
```

For whatever reason, this defaults to a `RuntimeError` which not really harder to catch - rails just doesn't. It's the difference between Exceptions and Errors in Ruby if you're familiar with that.